cns/ipampool/v2/monitor.go

```diff
@@ -0,0 +1,167 @@
+package v2
+
+import (
+	"context"
+	"math"
+	"sync"
+
+	"github.com/Azure/azure-container-networking/cns"
+	"github.com/Azure/azure-container-networking/crd/clustersubnetstate/api/v1alpha1"
+	"github.com/Azure/azure-container-networking/crd/nodenetworkconfig/api/v1alpha"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+const (
+	// DefaultMaxIPs default maximum allocatable IPs on a k8s Node.
+	DefaultMaxIPs = 250
+	// fieldManager is the field manager used when patching the NodeNetworkConfig.
+	fieldManager = "azure-cns"
+)
+
+type nodeNetworkConfigSpecUpdater interface {
+	PatchSpec(context.Context, *v1alpha.NodeNetworkConfigSpec, string) (*v1alpha.NodeNetworkConfig, error)
+}
```

Comment:
probably out of scope of this PR, but I would like to get us to a point where the ipampool doesn't know/care that we talk through kubernetes CRDs and this interface becomes something like an `IPRequester` which doesn't `PatchSpec` but instead `Do/Make/Apply` a domain type `Request`, and the `kubecontroller` package does the translation of the domain type to the k8s type

Reply:
moving in that direction, but the change to extract all of the data out of the NNC to feed the Monitor directly would be pretty invasive to the NNC reconciler and I'm not ready to do that in this change